src/components/PopupConfirmation.js

```diff
@@ -0,0 +1,32 @@
+import Popup from "./Popup.js";
+
+export default class PopupConfirmation extends Popup {
+  constructor(popupSelector, handleConfirm) {
+    super({ popupSelector });
+    this._handleConfirm = handleConfirm;
+    this._confirmButton = this._popupElement.querySelector(
+      ".modal__button-delete"
+    );
+  }
+
+  open(cardId) {
+    this._cardId = cardId; // store cardId when popup opens
+    super.open();
+  }
+
+  setConfirmCallback(callback) {
+    this._handleConfirm = callback;
+  }
+
+  close() {
+    super.close();
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._confirmButton.addEventListener("click", () => {
+      this._handleConfirm(this._cardId); // store cardId to delete handler
+      this.close();
```

Comment:
` this.close();` is not needed here

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**